backend/src/appointment/database/repo/schedule.py

```diff
@@ -69,3 +75,36 @@ def has_slot(db: Session, schedule_id: int, slot_id: int):
     """check if slot belongs to schedule"""
     db_slot = repo.slot.get(db, slot_id)
     return db_slot and db_slot.schedule_id == schedule_id
+
+
+def generate_slug(db: Session, schedule_id: int) -> str|None:
+    schedule = repo.schedule.get(db, schedule_id)
+
+    if schedule.slug:
+        return schedule.slug
+
+    # If slug isn't provided, give them the last 8 characters from a uuid4
+    # Try up-to-3 times to create a unique slug
+    for _ in range(3):
```

Comment:
I'm uhh not sure on a better way to do this right now.

Reply:
Yeah we always want a upper limit on these types of loop. So 3 is the limit here 😄 

They're also namespaced to the subscriber so it's very unlikely it'll ever create a duplicate.

Reply:
Agreed. If the 3 attempts are not sufficient, we should start asking general questions about randomness 😂